src/feasibility_checker.jl

```diff
@@ -0,0 +1,205 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+using LinearAlgebra
+
+"""
+    primal_feasibility_report(
+        model::Model,
+        [point::AbstractDict{VariableRef,Float64}];
+        atol::Float64 = 0.0,
+    )::Dict{Any,Float64}
+
+Given a dictionary `point`, which maps variables to primal values, return a
+dictionary mapping the constraint reference of each constraint in `model` to the
+distance between the point and the nearest feasible point, if the distance is
+greater than `atol`.
+
+## Notes
+
+ * If a variable is not given in `point`, the value is assumed to be `0.0`.
```

Comment:
I guess we could have
```Julia
primal_feasibility_report(model, point; skip_missing::Bool = false)
```
where constraints are skipped if `skip_missing=true`, otherwise the missing values default to `0`?

Reply:
Why is 0 a good default?

Reply:
`0` is a fairly conventional default. Especially for LP solvers that find basic solutions.

How about an argument: `default::Union{Float64,Missing} = 0`?

Reply:
Take a look at the latest commit. Sometimes I'm amazed at how Julia JustWorksᵀᴹ.